index.js

```diff
@@ -2699,6 +2700,29 @@
       [$.FiniteNumber, $.FiniteNumber, $.FiniteNumber],
       function(a, b) { return a + b; });
 
+  //# sum :: Foldable f => f FiniteNumber -> FiniteNumber
+  //.
+  //. Returns the sum of the given array of (finite) numbers.
+  //.
+  //. ```javascript
+  //. > S.sum([1, 2, 3, 4, 5])
+  //. 15
+  //.
+  //. > S.sum([])
+  //. 0
+  //.
+  //. > S.sum(S.Just(42))
+  //. 42
+  //.
+  //. > S.sum(S.Nothing())
+  //. 0
+  //. ```
+  var sum = S.sum =
+  def('sum',
+      {f: [Foldable]},
+      [f, $.FiniteNumber],
+      reduce_(function(a, b) { return a + b; }, 0));
```

Comment:
Might this be better?

```javascript
reduce(function(a) { return function(b) { return a + b; }; }, 0)
```

`reduce_` will curry the provided function, but we could remove a step by defining a curried function.

We needn't worry about this, though. Once we can express `f FiniteNumber` in the type system we'll be able to use `S.add` in this definition without references to that function appearing in error messages. :)